Client/Application/AppLaunchUtil.swift

```diff
@@ -83,27 +83,26 @@ class AppLaunchUtil {
 
     func setUpPostLaunchDependencies() {
 
-        let persistedCurrentVersion = InstallType.persistedCurrentVersion()
-        let introScreen = profile.prefs.intForKey(PrefsKeys.IntroSeen)
-        // upgrade install - Intro screen shown & persisted current version does not match
-        if introScreen != nil && persistedCurrentVersion != AppInfo.appVersion {
-            InstallType.set(type: .upgrade)
-            InstallType.updateCurrentVersion(version: AppInfo.appVersion)
-        }
-
+        // Ecosia: Removing unneded `introScreen` constant
+        // let introScreen = profile.prefs.intForKey(PrefsKeys.IntroSeen)
```

Comment:
Isn't `PrefsKeys.IntroSeen` still in use by the default browser promo? Is it safe to remove it from this logic?

Reply:
Indeed the `PrefsKeys.IntroSeen` is utilized for the default browser promo. However, the previous check was also confirming whether the `persistentCurrentVersion` was different from `AppInfo.appVersion` (see below)
```swift
introScreen != nil && persistedCurrentVersion != AppInfo.appVersion
```
Being specific, the `AppInfo.appVersion` has been hardcoded by Ecosia and always returning the `104.0` string. By doing so, the overall logic as per how it was, is overall irrelevant.

Reply:
You're totally right about the hardcoded `AppInfo.appVersion` and that part does seem irrelevant.

Was just mostly asking about the intro screen part since it is something from FF has which we make use of, seems even if was [somewhat updated](https://github.com/mozilla-mobile/firefox-ios/blob/a2a7bdf2aee75770dc9695553cc55bc7575230d6/Client/Application/AppLaunchUtil.swift#L96), at least comparing with what we have at the moment. But I agree if you say this was only relevant for their usage and not for our default browser promo.